engine/gamesys/src/gamesys/components/comp_collision_object.cpp

```diff
@@ -777,36 +888,18 @@ namespace dmGameSystem
             responsDDF.m_Normal = response.m_Normal;
             responsDDF.m_RequestId = request.m_UserId & 0xff;
 
-            message_id = dmPhysicsDDF::RayCastResponse::m_DDFDescriptor->m_NameHash;
-            descriptor = (uintptr_t)dmPhysicsDDF::RayCastResponse::m_DDFDescriptor;
-            data_size = sizeof(dmPhysicsDDF::RayCastResponse);
-            message_data = &responsDDF;
-        } else {
-            missedDDF.m_RequestId = request.m_UserId & 0xff;
-
-            message_id = dmPhysicsDDF::RayCastMissed::m_DDFDescriptor->m_NameHash;
-            descriptor = (uintptr_t)dmPhysicsDDF::RayCastMissed::m_DDFDescriptor;
-            data_size = sizeof(dmPhysicsDDF::RayCastMissed);
-            message_data = &missedDDF;
+            message_result = dmGameObject::PostDDF(&responsDDF, 0x0, resultReceiver, lua_listener, false);
         }
-
-        dmGameObject::HInstance instance = (dmGameObject::HInstance)request.m_UserData;
-        dmMessage::URL receiver;
-        receiver.m_Socket = dmGameObject::GetMessageSocket(dmGameObject::GetCollection(instance));
-        receiver.m_Path = dmGameObject::GetIdentifier(instance);
-        uint16_t component_index = request.m_UserId >> 16;
-        dmGameObject::Result result = dmGameObject::GetComponentId(instance, component_index, &receiver.m_Fragment);
-        if (result != dmGameObject::RESULT_OK)
+        else
         {
-            dmLogError("Error when sending ray cast response: %d", result);
+            dmPhysicsDDF::RayCastMissed missedDDF;
+            missedDDF.m_RequestId = request.m_UserId & 0xff;
+            message_result = dmGameObject::PostDDF(&missedDDF, 0x0, resultReceiver, lua_listener, false);
```

Comment:
To be able to call the callback many times we should use `dmGameObject::PostDDF` from `dmsdk/gameobject/script.h` instead of old `dmMessage::Post`